_stbt/core.py

```diff
@@ -1524,6 +1524,11 @@ def push_sample(self, sample):
                 color = (int(255 * max([1 - age, 0.5])),) * 3
                 _draw_text(img, x['text'], origin, color)
             for match_result in matches:
+                label = os.path.basename(match_result.image)
```

Comment:
I think this will fail if the image passed to `match` is a `numpy.array` rather than a filename.  `MatchResult`s are constructed as:

    result = MatchResult(..., image=(template.name or template.image))

Perhaps `_AnnotatedTemplate` should always have a non-`None` `.name` or similar?